tests/FSharp.Compiler.ComponentTests/ErrorMessages/UnsupportedAttributes.fs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.ComponentTests.ErrorMessages
+
+#if NETCOREAPP
+open Xunit
+open FSharp.Test.Utilities.Compiler
+
+module ``Unsupported Attributes`` =
+
+    [<Fact>]
+    let ``Warn successfully`` () =
+        """
+open System.Runtime.CompilerServices
+let f (w, [<CallerArgumentExpression "w">] x : string) = ()
+let [<ModuleInitializer>] g () = ()
+type C() =
+    member _.F (w, [<System.Runtime.CompilerServices.CallerArgumentExpression "w">] x : string) = ()
+    [<System.Runtime.CompilerServices.ModuleInitializer>]
+    member _.G() = ()
+        """
+        |> FSharp
+        |> typecheck
+        |> shouldFail
+        |> withResults [
```

Comment:
Just a note - a less verbose `withDiagnostics` can be used here, for example, like in the `ErrorMessages` suite:
https://github.com/dotnet/fsharp/blob/dbf9a625d3188184ecb787a536ddb85a4ea7a587/tests/FSharp.Compiler.ComponentTests/ErrorMessages/ClassesTests.fs#L68-L71

Reply:
@vzarytovskii How can I generate them? For `withResults` I can just copy from the test output.

Reply:
> @vzarytovskii How can I generate them? For `withResults` I can just copy from the test output.

I'm afraid, you'll have to write them manually :(
Let me see if I can serialise them into something which you can just copy-paste into `withDiagnostics`.